test.js

```diff
@@ -90,6 +90,20 @@ test('toVFile.read', function (t) {
     });
   });
 
+  t.test('should work in promise mode (buffer without encoding)', function (st) {
+    st.plan(3);
+
+    vfile.read('readme.md')
+      .then(function (file) {
+        st.equal(file.path, 'readme.md');
+        st.ok(buffer(file.contents));
+        st.equal(file.toString(), fixture);
+      })
+      .catch(function () {
+        t.fail('should resolve, not reject');
```

Comment:
Should this be `t` instead of `st`?

I dunno. The `plan` part will go wrong in any case, but I never really test promises this specifically.
What I sometimes do is make sure the “ok branch” tests are just as many as the “not ok branch”, by using something like this:

```js
st.deepEqual([file.path, buffer(file.contents), file.toString(), fixture], ['readme.md', true, fixture], 'should resolve')
```

What do you think?